File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main (void)
{   
    float change;                               //declare float variable for change                            
    do{ 
       change = get_float("Change owed: \n");   //prompt user for change
    }
    while(change < 0.009);                      //keep prompting user until they provide positive value, at least 1 cent
    int cents = round(change * 100);            //convert change into cents as integer
    int remainder = cents;                      //remainder will be the remaining change after giving out each coin
    int coins;                                  //coins is the number of coins - will accumulate over each while loop
    while (remainder >= 25){                    //repeat until max number of quarters are given
        coins = coins + 1;                      //increase coins count by 1
        remainder = remainder - 25;             //calculate remaining change
    }
    while(remainder >= 10){                     //repeat until max number of dimes are given
        coins = coins + 1;
        remainder = remainder - 10;
    }
    while(remainder >= 5){                      //repeat until max number of nickels are given
        coins = coins + 1;
        remainder = remainder - 5;
    }
    while(remainder >= 1){                      //repeat until max number of pennies are given
        coins = coins + 1;
        remainder = remainder - 1;
    }
    printf("I have %i coin(s).\n", coins);      //print number of coins given back
}

File: cash.py
from cs50 import get_float

# prompt user for change and make sure user provides a positive value
while True:
    change = get_float("Change owed: ")
    if change > 0.009:
        break

# convert change to cents, and round to the nearest cent
cents = round(change * 100)

# calculate number of coins and remainder based on Zamyla's % and // tip
# calculate number of quarters and remaining change
quarters = cents//25
remainder1 = cents % 25

# calculate number of dimes and remaining change
dimes = remainder1//10
remainder2 = remainder1 % 10

# calculate number of nickels and remaining change
nickels = remainder2//5
remainder3 = remainder2 % 5

# calculate number of pennies and remaining change
pennies = remainder3

# print number of coins for change, sum up various coins and round so it's an integer - voila!
print("I have", round(quarters + nickels + dimes + pennies), "coin(s).")